hack/verify-test-code.sh

```diff
@@ -46,6 +46,15 @@ do
     fi
 done
 
+errors_expect_no_equal=()
+for file in "${all_e2e_files[@]}"
+do
+    if grep -E "Expect\(.*\)\.(NotTo|ToNot)\((gomega\.Equal|Equal)" "${file}" > /dev/null
```

Comment:
we should really consider writing a better tool for this, piling on lots of grep over the k8s codebase has got to get slow...

how useful is enforcing this in CI, actually ...? 🤷‍♂

Reply:
This greps e2e test files only and even after applying this PR, this part (verify-test-code.sh) of pull-kubernetes-verify job takes 3 seconds as https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/89215/pull-kubernetes-verify/1240183762671636481
That doesn't seem to take much time and it would be fine to try finding a better way when it would take more time to run, I feel.